docs/root/configuration/http_filters/router_filter.rst

```diff
@@ -136,6 +141,13 @@ Note that retry policies can also be applied at the :ref:`route level
 
 By default, Envoy will *not* perform retries unless you've configured them per above.
 
+.. _config_http_filters_router_x-envoy-retriable-status-codes:
+
+x-envoy-retriable-status-codes
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Setting this header informs Envoy about what status codes should be considered retriable when used in
+conjunction with the `retriable-status-code` retry policy.
```

Comment:
How does this interact with the retry policy in terms of enablement? Do I need at least one such prefined retry code? Do we allow this header from untrusted clients? Etc.

Reply:
Added some more details to the comment